src/win/winpty.cc

```diff
@@ -277,22 +300,33 @@ static NAN_METHOD(PtyKill) {
   if (info.Length() != 2 ||
       !info[0]->IsNumber() ||
       !info[1]->IsNumber()) {
-    Nan::ThrowError("Usage: pty.kill(pid, innerPidHandle)");
+    Nan::ThrowError("Usage: pty.kill(pid, innerPid)");
     return;
   }
 
-  int handle = info[0]->Int32Value(Nan::GetCurrentContext()).FromJust();
-  HANDLE innerPidHandle = (HANDLE)info[1]->Int32Value(Nan::GetCurrentContext()).FromJust();
+  DWORD pid = info[0]->Uint32Value(Nan::GetCurrentContext()).FromJust();
+  DWORD innerPid = info[1]->Uint32Value(Nan::GetCurrentContext()).FromJust();
 
-  winpty_t *pc = get_pipe_handle(handle);
+  winpty_t *pc = get_pipe_handle(pid);
   if (pc == nullptr) {
     Nan::ThrowError("Pty seems to have been killed already");
     return;
   }
 
-  assert(remove_pipe_handle(handle));
+  assert(remove_pipe_handle(pid));
+
+  HANDLE innerPidHandle = OpenProcess(PROCESS_TERMINATE, FALSE, innerPid);
+  if (innerPidHandle == nullptr) {
+    return info.GetReturnValue().SetUndefined();
+  }
+  TerminateProcess(innerPidHandle, 0);
   CloseHandle(innerPidHandle);
 
+  // Finally close the handle of the spawned process.
+  HANDLE createdHandle = createdHandles[pid];
+  createdHandles.erase(pid);
+  CloseHandle(createdHandle);
```

Comment:
`createdHandles` holds handles of `innerPid` and not `pid` which corresponds to the winpty agent process.